controllers/Sources.php

```diff
@@ -165,7 +165,7 @@ public function write() {
         unset($data['ajax']);
 
         // check if source already exists
-        $id = \F3::get('PARAMS["id"]');
+        $id = intval(\F3::get('PARAMS["id"]'));
```

Comment:
This will break the source adding since `intval('new-123') === 0`, which is a valid ID according to `$sourcesDao->isValid`.